tests/observers/slack_resource_test.cpp

```diff
@@ -0,0 +1,63 @@
+#include "gmock/gmock.h"
+
+#include "mesos/mesos.hpp"
+#include "mesos/resources.hpp"
+
+#include "observers/slack_resource.hpp"
+
+#include "stout/try.hpp"
+#include "stout/nothing.hpp"
+
+#include "tests/common/sources/json_source.hpp"
+#include "tests/common/sinks/mock_sink.hpp"
+
+namespace mesos {
+namespace serenity {
+namespace tests {
+
+using ::testing::DoubleEq;
+using ::testing::Eq;
+using ::testing::InSequence;
+using ::testing::_;
+
+ACTION_P(BasicTestAction, check) {
+  Resources resources = arg0;
+
+  if (abs(check) < 0.0001) {
```

Comment:
That seems like it will be useful elsewhere.  What do you think about moving this to `tests/common`?  Maybe calling code could supply the tolerance.